### PR TITLE
Add support for CUDA 9.2 using gcc

### PR DESCRIPTION
**Pull request by BenjaminW3**
The error we have seen comes from the wrong placing of `ALPAKA_NO_HOST_ACC_WARNING` for one function. This seems to be supported by nvcc < 9.2.

### CODE REVIEW

**Comment by BenjaminW3** [comment]
Clang does not seem to work for now... I will investigate this in a separate issue.

**Comment by ax3l** [comment]
@BenjaminW3 can you rebase as well, so the other PR fixes are removed from the online diff?

**Comment by BenjaminW3** [comment]
I have already rebased. So it seems I have missed those newline changes in the previous PR 😭

**Comment by BenjaminW3** [comment]
I will move them into a separate PR.

**Comment by ax3l** [comment]
Nevermind, I was actually referring to another change but I mismatched two PRs ;-)

**Comment by BenjaminW3** [comment]
Ready to be merged! (I am not sure if the setting to remove approvals on each update is good for productivity)

File: include/alpaka/dev/cpu/SysInfo.hpp [hunk 1]
```diff
@@ -59,7 +59,7 @@ namespace alpaka
             namespace detail
             {
 #if BOOST_ARCH_X86
-    #if BOOST_COMP_GNUC || BOOST_COMP_CLANG || (!BOOST_COMP_MSVC_EMULATED && __INTEL_COMPILER)
+    #if BOOST_COMP_GNUC || BOOST_COMP_CLANG || (!BOOST_COMP_MSVC_EMULATED && defined(__INTEL_COMPILER))
```
Comment (psychocoderHPC) [inline include/alpaka/dev/cpu/SysInfo.hpp]: @BenjaminW3 
- Is this `defined` fix only an issue with CUDA 9.2 or is it an general issue? 
- Do we need this `defined` to have INTEL support or is is only a stile change?

**Pull request merged**